maxim/apis/maxim_apis.py

```
@@ -706,13 +706,24 @@ def add_dataset_entries(
         """
         try:
+            # Convert all entries to DatasetEntry objects for consistency
+            converted_entries = []
+            for entry in dataset_entries:
+                if isinstance(entry, DatasetEntry):
+                    converted_entries.append(entry)
+                elif isinstance(entry, dict):
+                    # Convert dictionary to DatasetEntry object
+                    converted_entries.append(DatasetEntry.from_json(entry))
+                else:
+                    raise TypeError(f"Invalid entry type: {type(entry).__name__}. Expected DatasetEntry or dict.")
```

Ruff (0.12.2)

Abstract `raise` to an inner function

(TRY301)